pkg/service/repair/worker.go

```diff
@@ -80,13 +79,7 @@ func (w *worker) Run(ctx context.Context) error {
 }
 
 func (w *worker) handleJob(ctx context.Context, job job) error {
-	var err error
-	if w.repairType == TypeRowLevel && job.Allowance.ShardsPercent == 0 {
-		err = w.rowLevelRepair(ctx, job)
-	} else {
-		err = w.legacyRepair(ctx, job)
-	}
-	return err
+	return w.rowLevelRepair(ctx, job)
```

Comment:
It is safe to drop the legacy partition level repair. All supported releases have row level repair now.